src/ui/public/visualize/visualize.js

```diff
@@ -133,9 +135,8 @@ uiModules
           if ($scope.editorMode) {
```

Comment:
Do we still need this `appState.vis` in there? Shouldn't that rather be something that the visualize editor directive now takes care about (and rather expose a callback we'll trigger every time we start a fetch?

Also why do we in general need to save the app state when we do a fetch?

Reply:
we need to propagate changes to the URL ...

i would prefer if visualize wouldn't know about appState but would rather receive a callback which it should call on props updates, and then visualize app can figure out if it wants to add it to url or not.

however i would suggest in a separate PR ?